### PR TITLE
Add support for 3D cuda grid

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

File: numba/cuda/cudaimpl.py [hunk 1]
```diff
@@ -56,6 +64,25 @@ def ptx_gridsize2d(context, builder, sig, args):
     return cgutils.pack_array(builder, [r1, r2])
 
 
+@register
+@implement('ptx.gridsize.3d', types.intp)
+def ptx_gridsize2d(context, builder, sig, args):
```
Comment (gmarkall) [inline numba/cuda/cudaimpl.py]: Although it's not called by name (IIUC), this should probably be renamed to `ptx_gridsize3d`.

File: numba/cuda/tests/cudapy/test_intrinsics.py [hunk 2]
```diff
@@ -150,6 +150,39 @@ def test_intrinsic_forloop_step(self):
...
 
+    def test_3dgrid(self):
+        @cuda.jit
+        def foo(out):
+            x, y, z = cuda.grid(3)
+            a, b, c = cuda.gridsize(3)
+            out[x, y, z] = a * b * c
+
+        arr = np.zeros(9 ** 3, dtype=np.int32).reshape(9, 9, 9)
+        foo[(3, 3, 3), (3, 3, 3)](arr)
```
Comment (pitrou) [inline numba/cuda/tests/cudapy/test_intrinsics.py]: Perhaps test with non-cube shapes?

Comment (pitrou) [inline numba/cuda/tests/cudapy/test_intrinsics.py]: Oops, sorry, it's already done below. My bad.

**Comment by pitrou** [comment]
This looks good to me.

**Pull request merged**